lib/job/repeat-every.js

```diff
@@ -13,7 +13,8 @@ module.exports = function(interval, options) {
   this.attrs.repeatInterval = interval;
   this.attrs.repeatTimezone = options.timezone ? options.timezone : null;
   if (options.skipImmediate) {
-    this.attrs.lastRunAt = new Date();
+    // Set the lastRunAt time to the nextRunAt so that the new nextRunAt will be computed in reference to the current value.
+    this.attrs.lastRunAt = this.attrs.nextRunAt;
```

Comment:
I believe this change belongs to the `computeNextRunAt()` function. It's better be implemented there not in this file.

Reply:
Thanks for the review! Could you elaborate? I'm not sure what you mean.
The `skipImmediate` option has always been calculated in `repeatEvery`, and `this.computeNextRunAt()` called afterwards to "skip" the immediate call, in order to avoid modifying `computeNextRunAt`. The change I made simply allows it to "remember" the previous `nextRunAt` value in the case of a chained call, like `job.schedule('in 3 days').repeatEvery('20 minutes')` (previously, the "in 3 days" part would be wiped by hard setting the lastRunAt to now). My change keeps the skipImmediate implementation within repeatEvery, as it was before.
Is your change request to move the `skipImmediate` implementation outside of repeatEvery and into computeNextRunAt entirely? If so, I think the most reasonable approach would be to keep track of a "skipNext" property in `attrs`. If true, `computeNextRunAt` would calculate an extra interval forward, and set skipNext to false. Would you agree?
If you wanted to keep the `repeatEvery` implementation, and just change `computeNextRunAt`, I'm not sure what you mean since the part where `lastRunAt` is set to now is the buggy part.

Reply:
Ok. I hear your point. You are right.
Let's make this line like this for safety reasons:

```js
this.attrs.lastRunAt = this.attrs.nextRunAt || new Date();
```

(Btw, unit tests are still failing.)

Reply:
Committed! Thank you!
In terms of unit tests, the "sets the nextRunAt property with skipImmediate" test is the one that is failing. I did edit this test. Before, it was calling `new Date()` with the `repeatEvery` call instead of the `new Job()` call. Due to my change, it uses the date defined in `new Job()` (as it should), so to check equality with that one, they must be declared at (about) the same time. 
I would guess that the Travis CI build is running on the old tests, and not my edited version.
Alternatively, it could be using my edited version, but there's a significant delay between `new Job()` and the next line, `new Date()`. That said, on my end, those tests pass just fine. Do you know how to get Travis to use my new tests, or how to figure out whether it is already?

It also looks like there's an additional failure on node versions 8 and 10 in the afterEach hook, however, that has nothing to do with this pull request and exists in the master branch and in the majority of recent pull requests.

Reply:
You would need to synchronise your branch with the latest `master`. You'd need to merge agenda's latest `master` to your branch.

Or start from scratch. There are not that many changes.

Reply:
The unit test would be fixed if you add that ` || new Date()` I suggested.

I don't know how to fix node.js 8 and 10 CI build. If you could do it - everyone would be happy! Including other contributors.

Reply:
I believe that instead of creating a new unit test you'd better amend this test: `'sets the nextRunAt property with skipImmediate'` to your needs.

Reply:
Thank you!
I went ahead and merged, everything should be passing now.